stable/sso-roles/templates/cluster-role-developers.yaml

```diff
@@ -0,0 +1,37 @@
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: developers
+rules:
+  # Full management of core API group resources, excluding create and delete operations
+  - apiGroups: [""]
+    resources: ["configmaps", "endpoints", "persistentvolumeclaims", "pods", "replicationcontrollers", "services", "serviceaccounts"]
+    verbs: ["get", "list", "watch", "update", "patch"]
+  # Management of batch resources, excluding create and delete operations
+  - apiGroups: ["batch"]
+    resources: ["cronjobs", "jobs"]
+    verbs: ["get", "list", "watch", "update", "patch"]
+  # Management of apps resources, excluding create and delete operations
+  - apiGroups: ["apps"]
+    resources: ["daemonsets", "deployments", "replicasets", "statefulsets"]
+    verbs: ["get", "list", "watch", "update", "patch"]
+  # Management of networking resources, excluding create and delete operations and network policies
+  - apiGroups: ["networking.k8s.io"]
+    resources: ["ingresses"]
+    verbs: ["get", "list", "watch", "update", "patch"]
+  # Read-only access to network policies
+  - apiGroups: ["networking.k8s.io"]
+    resources: ["networkpolicies"]
+    verbs: ["get", "list", "watch"]
+  # Access to view events to aid in debugging
+  - apiGroups: [""]
+    resources: ["events"]
+    verbs: ["get", "list", "watch"]
+  # Workload-specific access for custom resources, excluding potentially dangerous resources and restricting create and delete
+  - apiGroups: ["extensions", "argoproj.io", "aquasecurity.github.io", "external-secrets.io"]
```

Comment:
Not sure about these extensions, in which scenarios they might be needed for developers?

Reply:
removed